src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -218,8 +219,32 @@ public static JpegFileMarker FindNextFileMarker(byte[] marker, DoubleBufferedStr
         /// <typeparam name="TPixel">The pixel format.</typeparam>
         /// <param name="stream">The stream, where the image should be.</param>
         /// <returns>The decoded image.</returns>
-        public Image<TPixel> Decode<TPixel>(Stream stream)
+        public async Task<Image<TPixel>> DecodeAsync<TPixel>(Stream stream)
             where TPixel : unmanaged, IPixel<TPixel>
+        {
+            if (stream.CanSeek)
+            {
+                return this.Decode<TPixel>(stream);
+            }
+            else
+            {
+                using (var ms = new MemoryStream())
```

Comment:
In the future we may consider using our own variant of [RecyclableMemoryStream](https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream) to reduce allocation pressure.

Reply:
We don't even need one tbh. We know the length of the input stream, tightly control the lifecycle of the `MemoryStream`. We could simply rend an array from the shared pool and pass to the `MemoryStream` constructor.